jpegenc/subblocks/RLE/RLECore/entropycoder.py

```diff
@@ -8,31 +8,31 @@ def two2bin(num):
     return inum + 1
 
 
-def nbits(num, width):
+def nbits(num, width_data):
     """returns the number of bits required to store num"""
-    i = width - 1
+    i = width_data - 1
     while i >= 0:
         if num[i] == 1:
             return i + 1
         i = i - 1
-    return num[width]
+    return num[width_data]
 
 
 @block
-def entropycoder(width, clock, reset, data_in, size, amplitude):
+def entropycoder(width_data, clock, reset, data_in, size, amplitude):
```

Comment:
I don't think you need `width_data` and a parameter to this function (the parameters should come after the ports), or is width_data a port?  Is `width_data` not `len(data_in)`?